src/decoder_fmmidi.cpp

```diff
@@ -66,22 +69,34 @@ bool FmMidiDecoder::Open(FILE* file) {
 	}
 	seq->rewind();
 
-	ticks_per_sec = (float)seq->get_division() / tempo * 1000000;
+	tempo.emplace_back(this, midi_default_tempo);
 
 	return true;
 }
 
 bool FmMidiDecoder::Seek(size_t offset, Origin origin) {
+	assert(!tempo.empty());
+
 	if (offset == 0 && origin == Origin::Begin) {
 		mtime = seq->rewind_to_loop();
-		mtime_last_tempo_change = mtime_ltc_loopstart;
-		ticks_last_tempo_change = ticks_ltc_loopstart;
+
 		if (mtime > 0.0f) {
+			// Throw away all tempo data after the loop point
+			auto rit = std::find_if(tempo.rbegin(), tempo.rend(), [&](auto& t) { return t.mtime <= mtime; });
+			--rit;
+			auto it = std::next(rit).base();
```

Comment:
Why `--rit` and then `std::next(rit)` (which is just ++rit)? Maybe we just need to call `base()` directly? Again haven't tried this, so not sure what combination is needed to avoid off by 1 errors.

Reply:
thats just my lack of iter API knowledge. I searched for a std::cur() to call base on xD. This works.